### DISCUSSION TITLE
Getting REQUEST ENTITY TOO LARGE error on deploy, the requirements are fairly minimalistic

### DISCUSSION THREAD

**Discussion by manangupta19** [deta/cloud-docs#616] [Help]
I have the following dependencies in my requirements.txt: 
fastapi
numpy
opencv-python-headless
python-multipart

The other files are only a few KBs in size. The exact error is: 
Error: failed to update dependencies: Request entity too large

**Comment by sudo-subg** [comment]
Clone the micro using [deta clone](https://docs.deta.sh/docs/cli/commands/#deta-clone) in a new directory and check if there is any unnecessary `/tmp` directory created in it.

**Reply by manangupta19** [reply]
Did this. No /tmp directory created. Only what I had put there.

**Comment by AyushSehrawat** [comment]
Clone the repo and try to see if it has anything necessary, and also remove all the dependencies (empty requirements) , deploy, add again and deploy

**Reply by manangupta19** [reply]
I did as you suggested. It showed deployed changes successfully the first time. When I put in the requirements and deployed it showed the same error.

**Reply by sudo-subg** [reply]
As you have tried both the steps [above](https://github.com/orgs/deta/discussions/616#discussioncomment-4260898) and this one, it seems like the deployment size is getting over 250 MB.

**Reply by AyushSehrawat** [reply]
Try creating a new micro once

**Reply by manangupta19** [reply]
@AyushSehrawat created a new micro, still the same.

@mikBighne98 But that doesn't make sense. There are only two major dependencies but they don't go over 100MB combined and the other files aren't of a significant size either.

**Reply by sudo-subg** [reply]
Installed `opencv-python-headless` on my machine and it seems like after installation it takes almost 200 MB of storage and it installs `numpy` by default. So I removed `numpy` from requirements and deployed with `fastapi`, `opencv-python-headless`, `python-multipart` and it deployed successfully and logged that numpy was installed. So give it a try and let us know.

**Reply by manangupta19** [reply]
`opencv-python-headless` is what seems to be giving me the problem. I only put `fastapi` and `opencv-python-headless` in there and it's still giving an error. No idea what you are doing differently.

**Reply by sudo-subg** [reply]
![image](https://user-images.githubusercontent.com/102468247/204525678-5a50ae05-ad67-49df-80c2-a8b3a90c25b0.png)

**Reply by manangupta19** [reply]
Yeah, I don't even have python-multipart currently. Still not working. I am starting to give up on this and look for alternatives at this point.

**Comment by rohanshiva** [comment]
Please try creating a new micro with the following `requirements.txt`:

```
fastapi
python-multipart
```

The problem seems to be due to `opencv-python-headless`

**Reply by sudo-subg** [reply]
Then how it got deployed with my micro 🤔?

**Reply by manangupta19** [reply]
I need opencv for the api to work though, can't do it without.

**Reply by manangupta19** [reply]
@mikBighne98 are you on the most basic free plan?
